pennylane/beta/_qubit_device.py

```diff
@@ -0,0 +1,634 @@
+                are traced out of the system.
+
+        Returns:
+            array[float]: array of the resulting marginal probabilities.
+        """
+
+        if wires is None:
+            # no need to marginalize
+            return prob
+
+        wires = Wires(wires)
+        # determine which subsystems are to be summed over
+        inactive_wires = Wires.unique_wires([self.wires, wires])
+
+        # translate to wire labels used by device
+        device_wires = self.map_wires(wires)
+        inactive_device_wires = self.map_wires(inactive_wires)
+
+        # reshape the probability so that each axis corresponds to a wire
+        prob = self._reshape(prob, [2] * self.num_wires)
+
+        # sum over all inactive wires
+        prob = self._flatten(self._reduce_sum(prob, inactive_device_wires.labels))
+
+        # The wires provided might not be in consecutive order (i.e., wires might be [2, 0]).
+        # If this is the case, we must permute the marginalized probability so that
+        # it corresponds to the orders of the wires passed.
+        basis_states = np.array(list(itertools.product([0, 1], repeat=len(device_wires))))
+        perm = np.ravel_multi_index(
+            basis_states[:, np.argsort(np.argsort(device_wires))].T, [2] * len(device_wires)
+        )
+        return self._gather(prob, perm)
+
+    def expval(self, observable):
+
+        if self.analytic:
+            # exact expectation value
+            eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            prob = self.probability(wires=observable.wires)
+            return self._dot(eigvals, prob)
+
+        # estimate the ev
+        return np.mean(self.sample(observable))
+
+    def var(self, observable):
+
+        if self.analytic:
+            # exact variance value
+            eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            prob = self.probability(wires=observable.wires)
+            return self._dot((eigvals ** 2), prob) - self._dot(eigvals, prob) ** 2
+
+        # estimate the variance
+        return np.var(self.sample(observable))
+
+    def sample(self, observable):
+
+        # translate to wire labels used by device
+        device_wires = self.map_wires(observable.wires)
+        name = observable.name
+
+        if isinstance(name, str) and name in {"PauliX", "PauliY", "PauliZ", "Hadamard"}:
+            # Process samples for observables with eigenvalues {1, -1}
+            return 1 - 2 * self._samples[:, device_wires.labels[0]]
+
+        # Replace the basis state in the computational basis with the correct eigenvalue.
+        # Extract only the columns of the basis samples required based on ``wires``.
+        samples = self._samples[:, device_wires]
+        unraveled_indices = [2] * len(device_wires)
+        indices = np.ravel_multi_index(samples.T, unraveled_indices)
+        return observable.eigvals[indices]
+
+    def check_validity(self, queue, observables):
```

Comment:
This method is originally in `Device`, overwriting it here to avoid copying device to the beta folder.